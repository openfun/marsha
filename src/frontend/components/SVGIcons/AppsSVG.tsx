import React from 'react';

import SVGIcon, { SvgProps } from '.';

export const AppsSVG = (svgProps: SvgProps) => {
  return (
    <SVGIcon viewBox={{ x: 0, y: 0, width: 54, height: 54 }} {...svgProps}>
      <g fillRule="evenodd" stroke="none" strokeWidth="1">
        <g transform="translate(2.5 2.5)">
          <path
            fillRule="nonzero"
            d="M8.333 16.667h8.334V8.333H8.333v8.334zm12.5 25h8.334v-8.334h-8.334v8.334zm-12.5 0h8.334v-8.334H8.333v8.334zm0-12.5h8.334v-8.334H8.333v8.334zm12.5 0h8.334v-8.334h-8.334v8.334zm12.5-20.834v8.334h8.334V8.333h-8.334zm-12.5 8.334h8.334V8.333h-8.334v8.334zm12.5 12.5h8.334v-8.334h-8.334v8.334zm0 12.5h8.334v-8.334h-8.334v8.334z"
          ></path>
        </g>
      </g>
    </SVGIcon>
  );
};
