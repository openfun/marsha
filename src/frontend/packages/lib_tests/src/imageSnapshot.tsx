import path from 'path';
import { cleanup, render } from '@testing-library/react';
import { Grommet } from 'grommet';
import { generateImage, GenerateImageOptions } from 'jsdom-screenshot';
import { GlobalStyles, theme } from 'lib-common';
import React from 'react';

import { wrapInIntlProvider } from './intl';
import { wrapInRouter } from './router';

const checkSnapshot = () => {
  // check if test name contains '[screenshot]'
  if (!expect.getState().currentTestName?.includes('[screenshot]')) {
    throw new Error('[screenshot] is missing from test name');
  }
};

// To disable puppeteer sandbox if needed, launch the test suite
// using DISABLE_PUPPETEER_SANDBOX=1 as environment variable.
// see https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#setting-up-chrome-linux-sandbox
const disableSandbox = !!process.env.DISABLE_PUPPETEER_SANDBOX;

export const imageSnapshot = async (width?: number, height?: number) => {
  checkSnapshot();
  width = width || 800;
  height = height || 600;
  const generateImageOptions: GenerateImageOptions = {
    viewport: { width, height },
  };
  if (disableSandbox) {
    generateImageOptions.launch = {
      args: ['--no-sandbox', '--disable-setuid-sandbox'],
    };
  }
  const screenshot = await generateImage(generateImageOptions);
  expect(screenshot).toMatchImageSnapshot({
    customDiffDir: path.resolve(__dirname, '../../__diff_output__'),
    failureThreshold: 0.01,
    failureThresholdType: 'percent',
  });
};

export const renderImageSnapshot = async (
  component: JSX.Element,
  width?: number,
  height?: number,
) => {
  checkSnapshot();
  cleanup();
  render(
    wrapInIntlProvider(
      wrapInRouter(
        <Grommet theme={theme} style={{ height: '100%' }}>
          {component}
          <GlobalStyles />
        </Grommet>,
      ),
    ),
  );
  await imageSnapshot(width, height);
};

export const renderIconSnapshot = async (component: JSX.Element) => {
  checkSnapshot();
  await renderImageSnapshot(component, 100, 100);
};
