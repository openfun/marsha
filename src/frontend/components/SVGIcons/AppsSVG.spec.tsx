import React from 'react';
import { renderIconSnapshot } from 'utils/tests/imageSnapshot';

import { AppsSVG } from './AppsSVG';

describe('<AppsSVG />', () => {
  it('renders AppsSVG correctly [screenshot]', async () => {
    await renderIconSnapshot(<AppsSVG iconColor="#035ccd" />);
  });

  it('renders AppsSVG focus [screenshot]', async () => {
    await renderIconSnapshot(
      <AppsSVG iconColor="white" focusColor="blue-focus" />,
    );
  });
});
