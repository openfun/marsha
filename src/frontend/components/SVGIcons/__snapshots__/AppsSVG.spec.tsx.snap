// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AppsSVG correctly 1`] = `
.c0 path {
  fill: #035ccd;
}

<svg
  className="c0"
  viewBox="0 0 54 54"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fillRule="evenodd"
    stroke="none"
    strokeWidth="1"
  >
    <g
      transform="translate(2.5 2.5)"
    >
      <path
        d="M8.333 16.667h8.334V8.333H8.333v8.334zm12.5 25h8.334v-8.334h-8.334v8.334zm-12.5 0h8.334v-8.334H8.333v8.334zm0-12.5h8.334v-8.334H8.333v8.334zm12.5 0h8.334v-8.334h-8.334v8.334zm12.5-20.834v8.334h8.334V8.333h-8.334zm-12.5 8.334h8.334V8.333h-8.334v8.334zm12.5 12.5h8.334v-8.334h-8.334v8.334zm0 12.5h8.334v-8.334h-8.334v8.334z"
        fillRule="nonzero"
      />
    </g>
  </g>
</svg>
`;
